### PR TITLE
[13.0][PORT] #596 from 10.0: being able to make the difference between an update done by the customer and update done internally in Odoo

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
Port #596 from 10.0 to 13.0

### CODE REVIEW

**Comment by sebastienbeau** [comment]
@HviorForgeFlow  can you check why the bot is failling?

File: shopinvader/tests/test_cart.py [hunk 1]
```diff
@@ -99,6 +100,8 @@ class AnonymousCartCase(CartCase, CartClearTest):
     def setUp(self, *args, **kwargs):
         super(AnonymousCartCase, self).setUp(*args, **kwargs)
         self.cart = self.env.ref("shopinvader.sale_order_1")
+        self.cart.last_external_update_date = fields.Datetime.now()
+        time.sleep(1)
```
Comment (simahawk) [inline shopinvader/tests/test_cart.py]: these sleeps on every test, please use another approach like freezegun

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by rousseldenis** [comment]
@HviorForgeFlow

**Comment by rousseldenis** [comment]
@HviorForgeFlow Could you rebase ?

**Pull request closed without merging**